packages/core/src/typegenTypes.ts

```diff
@@ -22,19 +22,74 @@ export interface TypegenEnabled {
   '@@xstate/typegen': true;
 }
 export interface TypegenMeta extends TypegenEnabled {
-  matchesStates: string;
+  /**
+   * Allows you to specify all the results of state.matches
+   */
+  matchesStates: string | {};
```

Comment:
Added `{}` here so that we can pass `{ a: 'c' }` as the state.matches value

Reply:
Hm, `{}` might be somewhat "loose" since it doesn't quite mean "an empty object"  but IIRC we don't really even use this `TypegenMeta` that much (we might only use this in tests). It's just like a declaration and is used more as a documentation than implementation.

Reply:
Yeah I'm intentionally using the 'loose' aspect of it to allow us to pass anything resembling an object.